core/src/main/java/org/eobjects/datacleaner/extension/output/CreateCsvFileAnalyzer.java

```diff
@@ -103,20 +135,135 @@ public String getSuggestedLabel() {
 
     @Override
     public OutputWriter createOutputWriter() {
-        final String[] headers = new String[columns.length];
-        for (int i = 0; i < headers.length; i++) {
-            headers[i] = columns[i].getName();
+        List<String> headers = new ArrayList<String>() ;
+        for (int i = 0; i < columns.length; i++) {
+            String columnName = columns[i].getName();
+            headers.add(columnName);
+            if(columnToBeSortedOn != null) {
+                if(columnName.equals(columnToBeSortedOn.getName())){
+                    indexOfColumnToBeSortedOn = i ;
+                }
+            }
+        }
+
+        if(columnToBeSortedOn != null) {
+            if(indexOfColumnToBeSortedOn == -1) {
+                this.isColumnToBeSortedOnPresentInInput  = false ;
+                indexOfColumnToBeSortedOn = columns.length ;
+                headers.add(columnToBeSortedOn.getName()) ;
+                InputColumn<?>[] newColumns = new InputColumn<?>[columns.length + 1] ;
+                for(int i = 0; i < columns.length; i++){
+                    newColumns[i] = columns[i] ;
+                }
+                newColumns[columns.length]  = columnToBeSortedOn ;
+                columns = newColumns ;
+            }
         }
-        return CsvOutputWriterFactory.getWriter(file.getPath(), headers, separatorChar, quoteChar, escapeChar,
-                includeHeader, columns);
+       
+        
+        return CsvOutputWriterFactory.getWriter(_targetFile.getPath(), headers.toArray(new String[0]), separatorChar, quoteChar, escapeChar, includeHeader, columns);
     }
 
     @Override
     protected WriteDataResult getResultInternal(int rowCount) {
+        final CsvConfiguration csvConfiguration = new CsvConfiguration(CsvConfiguration.DEFAULT_COLUMN_NAME_LINE, FileHelper.DEFAULT_ENCODING,
+                separatorChar, quoteChar, escapeChar, false, true);
+
+        if (columnToBeSortedOn != null) {
+            
+            final CsvDataContext tempDataContext = new CsvDataContext(_targetFile, csvConfiguration);
+            final Table table = tempDataContext.getDefaultSchema().getTable(0);
+
+            final Comparator<? super Row> comparator = new Comparator<Row>() {
+                @SuppressWarnings("unchecked")
+                @Override 
+                public int compare(Row row1, Row row2) {
+                    Comparable<Object> value1 = (Comparable<Object>) row1.getValue(indexOfColumnToBeSortedOn);
+                    Comparable<Object> value2 = (Comparable<Object>) row2.getValue(indexOfColumnToBeSortedOn);
+                    int comparableResult = value1.compareTo(value2) ;
```

Comment:
Very minor: There's a risk of NPE here. Might want to use CompareUtils to avoid that.